src/openforms/submissions/logic/actions.py

```diff
@@ -225,16 +226,37 @@ class ServiceFetchAction(ActionOperation):
     def from_action(cls, action: ActionDict) -> "ServiceFetchAction":
         return cls(variable=action["variable"], fetch_config=action["action"]["value"])
 
-    def eval(self, context: DataMapping) -> DataMapping:
+    def eval(
+        self,
+        context: DataMapping,
+        log: Callable[[JSONValue], None],
+    ) -> DataMapping:
         # XXX perform_service_fetch is expressed in terms of FormVariables
         dummy_var = FormVariable(
             name=self.variable,
             service_fetch_configuration=ServiceFetchConfiguration.objects.get(
                 pk=self.fetch_config
             ),
         )
-        with log_errors({}, self.rule):  # TODO proper logging/error handling
-            return {dummy_var.name: perform_service_fetch(dummy_var, context)}
+        with log_errors({}, self.rule):  # TODO proper error handling
+            result = perform_service_fetch(dummy_var, context)
+            log(asdict(result))
```

Comment:
This is the most general thing that gives useful debugging information. I'm not sure if extra markup is wanted in the History view in  the admin.